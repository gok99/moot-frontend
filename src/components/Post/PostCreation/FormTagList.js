import React, { useEffect, useState } from 'react';
import { Row } from 'react-bootstrap';

import Tag from '../../Tag';

const FormTagList = (props) => {
  const [tagListState, setTagListState] = useState([]);
  useEffect(() => {
    setTagListState(
      props.postTagList.map((tag) => {
        return <Tag tagName={tag} owned={true} onTagPress={(tag) => props.onRemoveTag(tag)} postCreationCheck={true}/>
      })
    );
  });
  // const formTagList = ;

  return (
    <Row md="auto" className="b-taglist mt-2">
      {tagListState}
    </Row>
  );
};

export default FormTagList;