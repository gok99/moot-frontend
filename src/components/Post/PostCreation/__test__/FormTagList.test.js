import React from 'react';
import ReactDOM from 'react-dom';
import FormTagList from '../FormTagList';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<FormTagList postTagList={[]} onRemoveTag={(tag) => (event) => { event.preventDefault(); }} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<FormTagList postTagList={[]} onRemoveTag={(tag) => (event) => { event.preventDefault(); }} />).toJSON();
  expect(tree).toMatchSnapshot();
});