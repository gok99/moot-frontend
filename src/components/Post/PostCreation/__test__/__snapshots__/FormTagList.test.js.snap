// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="b-taglist mt-2 row row-cols-md-auto"
/>
`;
